src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+const body = document.querySelector('body');
+const div = document.createElement('div');
+let leftClickHappened = false;
+let rightClickHappened = false;
+
+div.setAttribute('data-qa', 'notification');
+body.appendChild(div);
+
+const showNotification = (message, type) => {
+  const notification = document.createElement('div');
+
+  notification.className = type;
+  notification.setAttribute('data-qa', 'notification');
+  notification.textContent = message;
+  body.appendChild(notification);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleLeftClick = (e) => {
+    if (e.button === 0) {
+      resolve(`First promise was resolved`);
+      document.removeEventListener('mouseup', handleLeftClick);
+    }
+  };
+
+  document.addEventListener('mouseup', handleLeftClick);
+
+  setTimeout(() => {
```

Comment:
The rejection handler in the setTimeout function should clear the event listener to avoid potential memory leaks or unintended behavior if the promise is rejected.